tests/framework/resourcemanager.go

```diff
@@ -809,3 +812,55 @@ func (rm *ResourceManager) WaitForGatewayObservedGeneration(
 		},
 	)
 }
+
+// GetNginxConfig uses crossplane to get the nginx configuration and convert it to JSON.
+func (rm *ResourceManager) GetNginxConfig(ngfPodName, namespace string) (*Payload, error) {
+	if err := injectCrossplaneContainer(
```

Comment:
Do we have to inject every time we get the nginx config? Can't we re-use the same crossplane container? Just wondering if this is going to be a pain if we have to get the nginx config multiple times in one test.

Reply:
Yeah I was going back and forth on this...part of my reasoning for putting it here was to encapsulate this process from a developers. I did put it in the `BeforeAll` at one point instead, but ran into an issue that I didn't look into further.

Reply:
Gotcha, we can leave it as-is for now and revisit if need be